### PR TITLE
1.7.0 release

### PR DESCRIPTION
**Pull request by ekes**
## Release Notes

We have a number of new features in this release:

### Sticky headers
If you want your site to have a sticky header when someone scrolls, you can check a checkbox now on the theme settings page. It has 3 options - not sticky, sticky, only sticky when scrolling.

### Better focus order for header
When tabbing through the header, the focus order is not more suited for screenreader users. Especially when you tab into and then out of the services menu.

### iOS Skip to Content link
We had a nasty issue which was very hard to replicate where keyboard users on iOS (iPhone, iPad) couldn't use the 'Skip to Content' link. Now they can.

### Back to top
We have a checkbox now to switch on/off a "Back to top" button which will appear after you scroll more than the height of your screen

### Accordion icons
Accordions now have a `+` icon when they are collapsed and a `-` icon when they are expanded

### Print mode
When printing a step-by-step or guide page now the navigation will not be printed.

### CODE REVIEW

**Comment by ekes** [comment]
Is there a dependency on `localgov` profile that should be declared now?

File: localgov_base.theme [hunk 1]
```diff
@@ -46,13 +46,52 @@ function localgov_base_form_system_theme_settings_alter(&$form, FormStateInterfa
...
+  ];
+
+}
+
+/**
+ * Implements hook_preprocess_html().
+ */
+function localgov_base_preprocess_html(&$variables) {
+  // Add the 'sticky-header' library if the sticky header setting is enabled.
+  if (theme_get_setting('localgov_base_header_behaviour') !== 'default') {
```
Comment (ekes) [inline localgov_base.theme:74]: Here, and above where it's used, in addition to telling people about the dependency, trying to enable it in both the profiles, we could also backstop for it not being there yet with:-

(theme_get_setting('localgov_base_header_behaviour') ?? 'default') !== 'default'

theme_get_setting returns NULL if the setting doesn't exist according to the docs https://api.drupal.org/api/drupal/core%21includes%21theme.inc/function/theme_get_setting/11.x

Comment (ekes) [reply localgov_base.theme:74]: This is safe I've tested it. It just evaluates `NULL !== 'default'` if it's not set.

**Comment by ekes** [comment]
= Blocker breaker

To release this these MRs should be merged at the same time:

For LocalGov:
 * https://github.com/localgovdrupal/localgov/pull/741

For LocalGov Microsites:
 * https://github.com/localgovdrupal/localgov_microsites_base/pull/279
 * https://github.com/localgovdrupal/localgov_microsites/pull/487
 
With these three https://github.com/localgovdrupal/localgov_base/issues/624 is resolved.

I have an open question if the tests will still pass, but that might just be my local https://github.com/localgovdrupal/localgov_base/issues/623

**Comment by ekes** [comment]
My notes about how to manually test (from Slack)

To review it all you'd have to:
 (a) install LocalGov (3.0.10)
 (b) upgrade the code of LocalGov to feature/740/enable-localgov_base-helper-module and LocalGov Base to 1.x
 (c) Run the updb
 (b) Make sure the localgov_base_helper module is enabled.
Then with the same code (LocalGov on feature/740/enable-localgov_base-helper-module and LocalGov Base on 1.x )
 (a) install LocalGov
 (b) Make sure the localgov_base_helper module is enabled.
Bonus for running phpunit on the test on localgov_base.
Next with microsites.
 (a) install LocalGov Microsites (4.0.0)
 (b) upgrade the code of LocalGov Microsites to feature/localgov-base-170 and LocalGov Microsites Base to feature/microsites-base-170 and LocalGov Base on 1.x
 (c) Run updb
 (d) Make sure the localgov_base_helper module is enabled.
Then with the same code (all those feature branches...)
 (a) install LocalGov Microsites
 (b) Make sure the localgov_base_helper module is enabled.
And finally a quick once over to check I wrote the composer.json version dependencies correctly.

**Comment by AWearring** [comment]
@ekes 
I was able to test (the first steps) on a copy of my local West Lindsey site and can confirm that `localgov_base_helper` is installed and enabled.

One thing I spotted on the "front end" is that https://github.com/localgovdrupal/localgov_base/pull/603 adds icons as expected, however this adds them to all sites _**even**_ if they have overwritten the Twig template to introduce their own icons:
image
Whilst the addition of the icons is good I don't think it is wise that these are added in using JavaScript (we also add the button using this method too) - I have a PR (https://github.com/localgovdrupal/localgov_paragraphs/pull/200) that @ctorgalson has reviewed and approved which moves the creation of the button and the addition of the icons into the Twig template which I think should be released along with localgov_base v1.7.0

**Comment by AWearring** [comment]
@ekes 
Also spotted these during `composer update`

Applying patches for drupal/preview_link
```
  https://www.drupal.org/files/issues/2024-05-28/3449121-9.patch (Add a 'copy to clipboard' feature for preview_link)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2024-05-28/3449121-9.patch
```
and
Applying patches for drupal/redirect
```
https://www.drupal.org/files/issues/2022-09-01/3057250-53.patch (Validation issue on adding url redirect: https://www.drupal.org/project/redirect/issues/3057250)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2022-09-01/3057250-53.patch
```

Looks like there are re-rolled patches available
- preview_link: https://www.drupal.org/project/preview_link/issues/3449121 | https://www.drupal.org/files/issues/2024-08-15/3449121-10.patch
- redirect: https://www.drupal.org/project/redirect/issues/3057250 | https://www.drupal.org/files/issues/2024-08-11/redirect--2024-08-11--3057250-79.patch

**Comment by ekes** [comment]
@AWearring:
> @ekes Also spotted these during `composer update`

Many thanks for the testing. I think both those patches are fixed in localgov (the profile) version 3.0.10

The https://github.com/localgovdrupal/localgov_base/issues/626 is a good catch and I think we want some resolution to it before pushing this out.

**Comment by stephen-cox** [comment]
The test failures I'm getting with this are:

```

1) Drupal\Tests\block_content\Functional\RegionRenderTest::testRegionRender with data set #0 ('disabled', 'FALSE')
Drupal\Core\Config\Schema\SchemaIncompleteException: Schema errors for localgov_base.settings with the following errors: localgov_base.settings:localgov_base_header_behaviour missing schema

/app/web/core/lib/Drupal/Core/Config/Development/ConfigSchemaChecker.php:98
/app/web/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php:111
/app/web/core/lib/Drupal/Core/Config/Config.php:230
/app/web/core/lib/Drupal/Core/Config/ConfigInstaller.php:396
/app/web/core/lib/Drupal/Core/Config/ConfigInstaller.php:149
/app/web/core/lib/Drupal/Core/ProxyClass/Config/ConfigInstaller.php:75
/app/web/core/lib/Drupal/Core/Extension/ThemeInstaller.php:251
/app/web/core/includes/install.core.inc:1654
/app/web/core/includes/install.core.inc:695
/app/web/core/includes/install.core.inc:572
/app/web/core/includes/install.core.inc:121
/app/web/core/lib/Drupal/Core/Test/FunctionalTestSetupTrait.php:315
/app/web/core/tests/Drupal/Tests/BrowserTestBase.php:570
/app/web/core/tests/Drupal/Tests/BrowserTestBase.php:370
/app/vendor/phpunit/phpunit/src/Framework/TestResult.php:729

2) Drupal\Tests\block_content\Functional\RegionRenderTest::testRegionRender with data set #1 ('sidebar_first', 'TRUE')
Drupal\Core\Config\Schema\SchemaIncompleteException: Schema errors for localgov_base.settings with the following errors: localgov_base.settings:localgov_base_header_behaviour missing schema

/app/web/core/lib/Drupal/Core/Config/Development/ConfigSchemaChecker.php:98
/app/web/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php:111
/app/web/core/lib/Drupal/Core/Config/Config.php:230
/app/web/core/lib/Drupal/Core/Config/ConfigInstaller.php:396
/app/web/core/lib/Drupal/Core/Config/ConfigInstaller.php:149
/app/web/core/lib/Drupal/Core/ProxyClass/Config/ConfigInstaller.php:75
/app/web/core/lib/Drupal/Core/Extension/ThemeInstaller.php:251
/app/web/core/includes/install.core.inc:1654
/app/web/core/includes/install.core.inc:695
/app/web/core/includes/install.core.inc:572
/app/web/core/includes/install.core.inc:121
/app/web/core/lib/Drupal/Core/Test/FunctionalTestSetupTrait.php:315
/app/web/core/tests/Drupal/Tests/BrowserTestBase.php:570
/app/web/core/tests/Drupal/Tests/BrowserTestBase.php:370
/app/vendor/phpunit/phpunit/src/Framework/TestResult.php:729

```

**Comment by millnut** [comment]
I wonder whether we revert PR https://github.com/localgovdrupal/localgov_base/pull/603 because of this issue https://github.com/localgovdrupal/localgov_base/issues/626 and revisit accordians again seperately

**Comment by finnlewis** [comment]
@millnut just chatting about your last comment in Merge Tuesday, and @AWearring and @markconroy this we can cover this in release notes.

**Pull request merged**